Lib/test/test_annotationlib.py

```diff
@@ -1675,6 +1675,24 @@ def test_name_lookup_without_eval(self):
 
         self.assertEqual(exc.exception.name, "doesntexist")
 
+    def test_evaluate_undefined_generic(self):
+        # Test the codepath where have to eval() with undefined variables.
```

Comment:
Can you use a test case that doesn't rely on direct construction of ForwardRef objects? The ForwardRef constructor is undocumented.

Reply:
Yep! I was using the constructor since the other `ForwardRef` tests also tend to use it, but I'll just rewrite it to use classes instead (as in the issue).